src/crypto/CHIPCryptoPALOpenSSL.cpp

```diff
@@ -1574,6 +1578,7 @@ CHIP_ERROR ValidateCertificateChain(const uint8_t * rootCertificate, size_t root
     X509_free(x509LeafCertificate);
     X509_free(x509CACertificate);
     X509_free(x509RootCertificate);
+    sk_X509_pop_free(chain, X509_free);
```

Comment:
Is this safe if early return done before push?

Reply:
Yes, it should be safe